common/src/main/java/org/wildfly/httpclient/common/HttpTargetContext.java

```diff
@@ -149,7 +149,7 @@ public void sendRequest(ClientRequest request, SSLContext sslContext, Authentica
     public void sendRequestInternal(final HttpConnectionPool.ConnectionHandle connection, ClientRequest request, AuthenticationConfiguration authenticationConfiguration, HttpMarshaller httpMarshaller, HttpResultHandler httpResultHandler, HttpFailureHandler failureHandler, ContentType expectedResponse, Runnable completedTask, boolean allowNoContent, boolean retry) {
         try {
             final boolean authAdded = retry || connection.getAuthenticationContext().prepareRequest(connection.getUri(), request, authenticationConfiguration);
-            request.getRequestHeaders().put(Headers.HOST, connection.getUri().getHost());
+            request.getRequestHeaders().put(Headers.HOST, connection.getUri().getHost() + ":" + connection.getUri().getPort());
```

Comment:
Is it made sure that the port is defined in this case? I would say it is better to add port only if it is defined and leave it without it if it isn't.